frontend/src/metabase/lib/form.tsx

```diff
@@ -0,0 +1,63 @@
+import React from "react";
+
+import { identity } from "lodash";
+import { BaseFieldDefinition } from "metabase-types/forms";
+import { ReactNode } from "react-markdown";
+
+export function getErrorMessageWithBoldFields(
+  message?: string,
+  formFields?: BaseFieldDefinition[],
+): ReactNode | undefined {
+  if (!message) {
+    return undefined;
+  }
+
+  const fieldTitles = (formFields || [])
+    .map(formField => formField.title)
+    .filter(identity) as string[];
+
+  if (fieldTitles.length === 0) {
+    return message;
+  }
+
+  const tokenizedMessage = tokenizeErrorMessage(message, fieldTitles);
+  const hasBoldFields = tokenizedMessage.length > 1;
+  if (hasBoldFields) {
+    return tokenizedMessage;
+  } else {
+    return message;
+  }
+}
+
+function tokenizeErrorMessage(
```

Comment:
@ranquild Could you try having a quick look if the new implementation of `tokenizeErrorMessage` looks alright now? This is the only function I've changed in this file. The rest are the same.

Reply:
@WiNloSt Looks good